### PR TITLE
[JENKINS-72579] Adjust heap dump file name for compatibility with OpenJDK file suffix requirements

### PR DESCRIPTION
**Pull request by balaji-sivasakthi**
See [JENKINS-72579](https://issues.jenkins.io/browse/JENKINS-72579)

### Testing done
Done 

### Proposed changelog entries

- Adjust heap dump file name for compatibility with OpenJDK file suffix requirements.

### Proposed upgrade guidelines
N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/hudson/util/RemotingDiagnostics.java
Comment (mawinter69) [inline core/src/main/java/hudson/util/RemotingDiagnostics.java:218]: wouldn't it be easier to just change the suffix to `.hprof` in line 179?

Comment (balaji-sivasakthi) [reply core/src/main/java/hudson/util/RemotingDiagnostics.java]: ok, I changed  it.

**Comment by basil** [comment]
Software testing is necessary to identify and fix any errors or bugs and to ensure the software functions as expected, providing a reliable and user-friendly experience. For this reason, the **Testing Done** section of the PR template is mandatory, not optional. Can you please confirm that you can start Jenkins with `java -jar jenkins.war`, complete the setup wizard, and visit `http://${JENKINS_URL}/heapDump` before and after this change. Before this PR you should get an error, and after this PR your browser should download an hprof heap dump which you should be able to open with e.g. VisualVM. If you can complete all this testing as expected then please update the **Testing Done** section with the results.

**Comment by balaji-sivasakthi** [comment]
Before without `.` 

```
java.lang.IllegalArgumentException: heapdump file must have .hprof extension
        at jdk.management/com.sun.management.internal.HotSpotDiagnostic.dumpHeap(HotSpotDiagnostic.java:60)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:64)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at java.base/sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:97)
        at java.management/com.sun.jmx.mbeanserver.ConvertingMethod.invokeWithOpenReturn(ConvertingMethod.java:193)
        at java.management/com.sun.jmx.mbeanserver.ConvertingMethod.invokeWithOpenReturn(ConvertingMethod.java:175)
        at java.management/com.sun.jmx.mbeanserver.MXBeanIntrospector.invokeM2(MXBeanIntrospector.java:115)
        at java.management/com.sun.jmx.mbeanserver.MXBeanIntrospector.invokeM2(MXBeanIntrospector.java:52)
        at java.management/com.sun.jmx.mbeanserver.MBeanIntrospector.invokeM(MBeanIntrospector.java:236)
        at java.management/com.sun.jmx.mbeanserver.PerInterface.invoke(PerInterface.java:138)
        at java.management/com.sun.jmx.mbeanserver.MBeanSupport.invoke(MBeanSupport.java:252)
        at java.management/javax.management.StandardMBean.invoke(StandardMBean.java:405)
        at
```

After including `.`
Screenshot 2024-01-20 at 11 17 56 AM

**Review by mawinter69 — APPROVED** [review]

**Review by basil — APPROVED** [review]
Looks good. Should include a changelog entry I think.

**Comment by balaji-sivasakthi** [comment]
@basil, should I add it, or will the core team add it? If I should add it, how do I do so?

**Comment by basil** [comment]
> If I should add it, how do I do so?

You can edit the PR description to fill in the "Proposed changelog entries" section.

**Comment by balaji-sivasakthi** [comment]
@basil I have updated the changelog.

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
